review/src/NoDuplicatePorts.elm

```diff
@@ -0,0 +1,114 @@
+module NoDuplicatePorts exposing (rule)
+
+{-| Highlights duplicate port names, which will break the Elm compiler
+-}
```

Comment:
```suggestion
{-|

@docs rule

-}
```

If you want to publish this, `rule` will need documentation. Instead of having everything duplicated, I think it's more interesting to have the documentation on the rule than on the module.

In the rule's documentation, you could then have something like

```elm
{-| Ensure that port names are unique across your project.

Problem: When there are multiple ports with the same name, the Elm compiler... blabla TODO

You should not enable this rule when the project is an Elm package.
-}
rule : Rule
```